### DISCUSSION TITLE
How to use correct types when using the `createAction` helper?

### DISCUSSION THREAD

**Discussion by johannesjo** [ngrx/platform#3121] [Q&A] [answered]
There are places in my app (e.g. some effects and meta reducers) where I'd like to manually specify the action type. Before transitioning to the action creator functions I simply could use the action class at those places, but this is not possible any more. The only thing I managed to do is this:
```
const myActionAsVarOrFunctionParam: ({
          myPayload: {myPayloadProp: string};
        } & TypedAction<'ACTION_TYPE'>) =  myAction(...); 
```

Which clearly isn't helpful as it has no relation to the stuff defined with `createAction()`. Is there some way to derive or at least share a type when using the action creators? So something like this:
```
type MySharedType =  ({
          myPayload: {myPayloadProp: string};
        } & TypedAction<'ACTION_TYPE'>);

export const myAction: MySharedType = createAction(...);
```

**Comment by timdeschryver** [comment, marked as answer]
I think `ReturnType<typeof actionName>` is what you're looking for

**Reply by johannesjo** [reply]
This is great! Thank you very much!
